organization/models.py

```diff
@@ -2,105 +2,153 @@
+        return group_members
+
+    @property
+    def active_members_count(self) -> int:
+        return len(self.active_members)
 
     def __str__(self):
         return "Group %s" % self.name
 
     def __repr__(self):
         return "Group(name=%s)" % self.name
 
-    class Meta:
-        default_related_name = 'groups'
-        verbose_name_plural = 'Internal groups'
+
+class InternalGroupPositionMembership(models.Model):
+    """
+    An intermediary model between a user and a InternalGroupPosition with additional information
+    regarding membership
+    """
+    date_joined = models.DateField(auto_now_add=True)
+    date_ended = models.DateField(
+        default=None,
+        null=True,
+        blank=True
+    )
+
+    user = models.ForeignKey("users.User", on_delete=models.CASCADE)
+    position = models.ForeignKey("organization.InternalGroupPosition", related_name="memberships",
+                                 on_delete=models.DO_NOTHING)
 
 
 class InternalGroupPosition(models.Model):
     """
     A position for an internal group, e.g. Hovmester
     """
-    STATUS = KSG_POSITIONS
 
-    name = StatusField(unique=True)
+    class Meta:
+        unique_together = ("name", "internal_group",)
 
-    description = models.CharField(max_length=1024, blank=True, null=True)
+    # feels natural to have the type here, like functionary or gang-member, but how do we deal with it as an
+    # interest group? Do we just have null=True and then not set it?
+    # Then again we have to make distinctions between being active/retired funk/gjeng or a hangaround. Just
+    # say fuckit and keep it as-is in the users.User model? Solve it at business-logic level through views?
 
+    name = models.CharField(unique=True, max_length=32)
+    internal_group = models.ForeignKey(InternalGroup, null=False, blank=False, on_delete=models.CASCADE,
+                                       related_name="positions")
+    description = models.CharField(max_length=1024, blank=True, null=True)
     holders = models.ManyToManyField(
         User,
-        blank=True,
-        related_name='positions'
+        related_name='positions',
+        through="organization.InternalGroupPositionMembership"
     )
 
+    @property
+    def active_memberships(self):
+        return self.memberships.filter(date_ended__isnull=True)
+
+    @property
+    def active_memberships_count(self) -> int:
+        return self.active_memberships.count()
+
     def __str__(self):
         return "Position %s" % self.name
 
     def __repr__(self):
         return "Position(name=%s)" % self.name
+
+
+class Commission(models.Model):
+    """
+    A commissions (verv) in KSG. A commissions can be shared by many users (e.g. Personal),
+    or created specifically for this internal group (e.g. Hybelbarsjef).
+    """
+
+    name = models.CharField(max_length=32, unique=True)
+    manager = models.ForeignKey(  # does this thing even make sense to have?
```

Comment:
The idea is that a single person should manage the commission?

Reply:
I suppose in some cases where it makes sense, but not very happy with this approach. Think I'll scrap this field for the time being.